### PR TITLE
Added ability to include hidden files in sync

### PR DESCRIPTION
**Pull request by pierskarsenbarg**
Fixes: #12

### CODE REVIEW

File: examples/synced-folder-aws-csharp/Program.cs [hunk 1]
```diff
@@ -14,5 +14,6 @@
         BucketName = bucket.BucketName,
         Acl = (string)CannedAcl.PublicRead,
         ManagedObjects = false,
+        IncludeHiddenFiles = true
```
Comment (cnunciato) [inline examples/synced-folder-aws-csharp/Program.cs:17]:
```suggestion
        IncludeHiddenFiles = true,
```

File: examples/synced-folder-aws-python/__main__.py [hunk 2]
```diff
@@ -11,4 +11,5 @@
     path="./my-folder",
     bucket_name=bucket.bucket,
     acl=s3.CannedAcl.PUBLIC_READ,
+    include_hidden_files=True
```
Comment (cnunciato) [inline examples/synced-folder-aws-python/__main__.py:14]:
```suggestion
    include_hidden_files=True,
```

File: examples/synced-folder-aws-typescript/index.ts [hunk 3]
```diff
@@ -9,4 +9,5 @@ const folder = new synced.S3BucketFolder("synced-folder", {
     path: "./my-folder",
     bucketName: bucket.bucket,
     acl: aws.s3.PublicReadAcl,
+    includeHiddenFiles: true
```
Comment (cnunciato) [inline examples/synced-folder-aws-typescript/index.ts:12]:
```suggestion
    includeHiddenFiles: true,
```

File: examples/synced-folder-aws-yaml/Pulumi.yaml [hunk 4]
```diff
@@ -14,3 +14,4 @@ resources:
       path: ./my-folder
       bucketName: ${my-bucket.bucket}
       acl: public-read
+      includeHiddenFiles: true
```
Comment (cnunciato) [inline examples/synced-folder-aws-yaml/Pulumi.yaml:17]:
```suggestion
      includeHiddenFiles: true,
```

File: examples/synced-folder-azure-python/__main__.py [hunk 5]
```diff
@@ -23,4 +23,5 @@
     storage_account_name=account.name,
     container_name=container.name,
     path="./my-folder",
+    include_hidden_files=True
```
Comment (cnunciato) [inline examples/synced-folder-azure-python/__main__.py:26]:
```suggestion
    include_hidden_files=True,
```

File: examples/synced-folder-azure-typescript/index.ts [hunk 6]
```diff
@@ -22,4 +22,5 @@ const folder = new synced.AzureBlobFolder("synced-folder", {
     storageAccountName: account.name,
     containerName: container.name,
     path: "./my-folder",
+    includeHiddenFiles: true
```
Comment (cnunciato) [inline examples/synced-folder-azure-typescript/index.ts:25]:
```suggestion
    includeHiddenFiles: true,
```

File: examples/synced-folder-gcp-python/__main__.py [hunk 7]
```diff
@@ -19,5 +19,6 @@
     pulumi_synced_folder.GoogleCloudFolderArgs(
         bucket_name=bucket.name,
         path="./my-folder",
+        include_hidden_files=True
```
Comment (cnunciato) [inline examples/synced-folder-gcp-python/__main__.py:22]:
```suggestion
        include_hidden_files=True,
```

File: examples/synced-folder-gcp-typescript/index.ts [hunk 8]
```diff
@@ -16,4 +16,5 @@ const binding = new gcp.storage.BucketIAMBinding("binding", {
 const folder = new synced.GoogleCloudFolder("folder", {
     bucketName: bucket.name,
     path: "./my-folder",
+    includeHiddenFiles: true
```
Comment (cnunciato) [inline examples/synced-folder-gcp-typescript/index.ts:19]:
```suggestion
    includeHiddenFiles: true,
```

File: provider/cmd/pulumi-resource-synced-folder/azure-blob-folder.ts [hunk 9]
```diff
@@ -34,7 +35,7 @@ export class AzureBlobFolder extends pulumi.ComponentResource {
         args.managedObjects = args.managedObjects ?? true;
         args.disableManagedObjectAliases = args.disableManagedObjectAliases ?? false;
```
Comment (cnunciato) [inline provider/cmd/pulumi-resource-synced-folder/azure-blob-folder.ts:36]: I like being explicit about these defaults -- okay with this, to align with others?

```suggestion
        args.disableManagedObjectAliases = args.disableManagedObjectAliases ?? false;
        args.includeHiddenFiles = args.includeHiddenFiles ?? false;
```

File: provider/cmd/pulumi-resource-synced-folder/google-cloud-folder.ts [hunk 10]
```diff
@@ -32,7 +33,7 @@ export class GoogleCloudFolder extends pulumi.ComponentResource {
         args.managedObjects = args.managedObjects ?? true;
         args.disableManagedObjectAliases = args.disableManagedObjectAliases ?? false;
```
Comment (cnunciato) [inline provider/cmd/pulumi-resource-synced-folder/google-cloud-folder.ts:34]:
```suggestion
        args.disableManagedObjectAliases = args.disableManagedObjectAliases ?? false;
        args.includeHiddenFiles = args.includeHiddenFiles ?? false;
```

File: provider/cmd/pulumi-resource-synced-folder/s3-bucket-folder.ts [hunk 11]
```diff
@@ -33,7 +34,7 @@ export class S3BucketFolder extends pulumi.ComponentResource {
         args.managedObjects = args.managedObjects ?? true;
         args.disableManagedObjectAliases = args.disableManagedObjectAliases ?? false;
```
Comment (cnunciato) [inline provider/cmd/pulumi-resource-synced-folder/s3-bucket-folder.ts:35]:
```suggestion
        args.disableManagedObjectAliases = args.disableManagedObjectAliases ?? false;
        args.includeHiddenFiles = args.includeHiddenFiles ?? false;
```

File: schema.json [hunk 12]
```diff
@@ -37,6 +37,10 @@
                 "disableManagedObjectAliases": {
                     "type": "boolean",
                     "description": "Disables adding an [alias](https://www.pulumi.com/docs/intro/concepts/resources/options/aliases/) resource option to managed objects in the bucket."
+                },
+                "includeHiddenFiles": {
+                    "type": "boolean",
+                    "description": "Include hidden files (\"dotfiles\") when synchronixing folders. False by default."
```
Comment (cnunciato) [inline schema.json:43]: Just to align with the others:

```suggestion
                    "description": "Include hidden files (\"dotfiles\") when synchronizing folders. Defaults to `false`."
```

File: schema.json [hunk 13]
```diff
@@ -77,6 +81,10 @@
...
```
Comment (cnunciato) [inline schema.json:87]:
```suggestion
                    "description": "Include hidden files (\"dotfiles\") when synchronizing folders. Defaults to `false`."
```

File: schema.json [hunk 14]
```diff
@@ -110,6 +118,10 @@
...
```
Comment (cnunciato) [inline schema.json:124]:
```suggestion
                    "description": "Include hidden files (\"dotfiles\") when synchronizing folders. Defaults to `false`."
```

**Review by cnunciato — COMMENTED** [review]
This looks great, thanks! Only a few little nits.

Also, I'm not sure how others do it, but would you mind leaving out the SDK build artifacts? That way we can do the build, test, and tag separate from any particular PR.

File: schema.json (hunk 12 above)
Comment (pierskarsenbarg) [reply schema.json:43]: I can't even blame this on US/UK spelling differences 🤦

File: provider/cmd/pulumi-resource-synced-folder/s3-bucket-folder.ts (hunk 11 above)
Comment (pierskarsenbarg) [reply provider/cmd/pulumi-resource-synced-folder/s3-bucket-folder.ts:35]: Shall I still leave the default value in the function signature?

File: provider/cmd/pulumi-resource-synced-folder/azure-blob-folder.ts (hunk 9 above)
Comment (pierskarsenbarg) [reply provider/cmd/pulumi-resource-synced-folder/azure-blob-folder.ts:36]: You shouldn't need this because the function parameter has a default value

**Comment by pierskarsenbarg** [comment]
@cnunciato I think it was Luke that suggested that we check in the SDK folder as it means we can track the SDKs at a particular point in time

**Comment by cnunciato** [comment]
@pierskarsenbarg Ok sounds good -- I'm happy to adhere to whatever the conventions are. Fo feel free to leave 'em in, then, and regenerate when you're ready for a final review.

**Comment by pierskarsenbarg** [comment]
@cnunciato I've checked the SDKs in. All ready for final review.

**Review by cnunciato — APPROVED** [review]

**Pull request merged**